ensime_shared/ensime.py

```diff
@@ -136,16 +136,23 @@ def queue_poll(self, sleep_t=0.5):
 
         Value of sleep is low to improve responsiveness.
         """
+        connection_alive = True
```

Comment:
I'm not totally sure why this is necessary. From my reading of the code, `teardown()` will flip `self.running` to `False`, which should terminate the loop on the next pass. So this saves a 1/2 second sleep, but that's about it. 

Am I correct in this interpretation or is there something I'm missing?